.github/workflows/build.yml

```diff
@@ -5,63 +5,74 @@ on:
       - main
 
 jobs:
-  build:
+  get-version:
     runs-on: ubuntu-latest
-    permissions:
-      contents: write
-      id-token: write
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-      - name: Install Node.js
-        uses: actions/setup-node@v4
-        with:
-          node-version: 20
       - name: Get version
         id: get_version
         run: echo "version=$(grep "version" Cargo.toml | head -n1 | cut -d '"' -f2)" >> $GITHUB_OUTPUT
       - name: Compare version to NPM
         run: |
-          NPM_VERSION=$(npm view create-o7-app version)
+          NPM_VERSION=$(curl -s https://registry.npmjs.org/create-o7-app | jq -r '.["dist-tags"].latest')
           MY_VERSION=${{ steps.get_version.outputs.version }}
           if [ "$NPM_VERSION" == "$MY_VERSION" ]; then
             echo "NPM version is the same as the version in Cargo.toml, exiting"
             exit 1
           fi
-      - name: Install pnpm
-        run: npm i -g pnpm
-
-      - name: Generate changelogs
-        run: |
-          # Full changelog
-          pnpx git-cliff --tag ${{ steps.get_version.outputs.version }} -o CHANGELOG.md
-          # Changelog for the release
-          pnpx git-cliff --tag ${{ steps.get_version.outputs.version }} --unreleased --strip header -o CHANGELOG-release.md
 
+  build-linux:
+    needs: get-version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
         with:
-          components: rustfmt
-          targets: x86_64-unknown-linux-musl,x86_64-pc-windows-gnu,i686-pc-windows-gnu,x86_64-apple-darwin,aarch64-apple-darwin
-      # - name: Test
-      #   run: cargo test
-      - name: Install MinGW
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y gcc-mingw-w64-x86-64 gcc-mingw-w64-i686
+          targets: x86_64-unknown-linux-musl
+      - name: Cache Rust
```

Comment:
Can we run this once as part of the get-version job and then restore the cache for each action?

Reply:
In the get-version job, we can only cache the download of dependencies. By adding caching to each job, we can save the build of these dependencies, significantly speeding up the process.

Reply:
Per [this](https://github.com/Swatinem/rust-cache?tab=readme-ov-file#cache-details) it looks like it's keyed off the Cargo.lock, which will change almost every single build because I have it auto-updating cargo dependencies

Reply:
I understand, but in practice the cache of the first job will also change in each version, since the build jobs happen in parallel the final time should be the same. I think applying the cache to the entire build might be more interesting.
Perhaps a way to have the best of both worlds is to have two different caches, one for downloading the dependencies and another for the final build. I'll test this later, if it doesn't make much sense I can keep only the download cache.

Reply:
Yeah but my point is the first job prevents all 5 builds from downloading, while the cache in each build won't be reused

Reply:
Ok, I'll just leave the download cache.

I still think the build cache would be interesting, it would be reused in cases of PRs with changes only in the code or in the templates, such as the case of automatic template updates.

Reply:
I was confusing the build and test workflows, it really doesn't make sense to cache the build.

Regarding the download cache, I did some tests and it actually slows down the process.

- get-version job: 12s to install rust, init cache and post cache
- build jobs: 2s to download cache and post cache
- cargo fetch: 4s

Only 2s saved in each build job, while spending 16s on the cache setup.

with cache: https://github.com/ErickCReis/create-o7-app/actions/runs/10119178428/job
without cache: https://github.com/ErickCReis/create-o7-app/actions/runs/10119247497

I think it would be possible to leave it without cache for now and if the project has more dependencies or the build process happens at other times we can check again if the cache actually improves the time, what do you think?

Reply:
I think no cache is fine, especially since after #59 took build times from 14 -> 3 minutes. Curious if parallelizing still is significant now with the overhead of spinning up more runners.

Reply:
I hadn't noticed, it actually got a lot faster now.

Here's the latest compilation, it took less than 2 minutes, it's better but I don't know if it's really worth it.
https://github.com/ErickCReis/create-o7-app/actions/runs/10124568557

I updated the PR. If it's not worth it, feel free to close it.

Reply:
I think I'll close it until the build times become too big again, the test parallelization is going to be much more impactful